upstairs/src/upstairs.rs

```diff
@@ -1143,54 +1239,91 @@ impl Upstairs {
     }
 
     /// Submits a new write job to the upstairs
-    async fn submit_write(
+    ///
+    /// This function **defers** the write job submission, because writes
+    /// require encrypting data (which is expensive) and we'd like to return as
+    /// quickly as possible.
+    fn submit_deferred_write(
         &mut self,
         offset: Block,
         data: Bytes,
         req: BlockReq,
         is_write_unwritten: bool,
     ) {
-        self.submit_write_inner(offset, data, Some(req), is_write_unwritten)
-            .await
+        self.submit_deferred_write_inner(
+            offset,
+            data,
+            Some(req),
+            is_write_unwritten,
+        )
     }
 
     /// Submits a dummy write (without an associated `BlockReq`)
+    ///
+    /// This **does not** go through the deferred-write pipeline
     #[cfg(test)]
     pub(crate) async fn submit_dummy_write(
         &mut self,
         offset: Block,
         data: Bytes,
         is_write_unwritten: bool,
     ) {
-        self.submit_write_inner(offset, data, None, is_write_unwritten)
-            .await
+        if let Some(w) = self
+            .compute_deferred_write(offset, data, None, is_write_unwritten)
+            .and_then(DeferredWrite::run)
+        {
+            self.apply_guest_request(DeferredBlockReq::Write(w)).await
+        }
     }
 
     /// Submits a new write job to the upstairs, optionally without a `BlockReq`
     ///
     /// # Panics
     /// If `req` is `None` and this isn't running in the test suite
-    async fn submit_write_inner(
+    fn submit_deferred_write_inner(
         &mut self,
         offset: Block,
         data: Bytes,
         req: Option<BlockReq>,
         is_write_unwritten: bool,
     ) {
+        // It's possible for the write to be invalid out of the date, in which
```

Comment:
Is this a typo here?  Maybe `invalid out of the gate`?
Or, something else?

Reply:
Yup, this was meant to be "out of the gate".  Fixed in 9c836a3d1dc7d056b619b61210e99866706c3867